samples/python/reactors/PorousMediaBurner.py

```diff
@@ -0,0 +1,395 @@
+
+        for i in range(self.n_vars):
+            LHS[i] = 1.0
+            RHS[i] = 0.0
+
+        # update solid properties with current values of the solid temperature
+        self.solid.solid_phase.TP = self.Ts, constantP
+
+        # create some variables for convenience
+        density = self.thermo.density_mass
+        cp = self.thermo.cp_mass
+        porosity = self.solid.porosity
+        solid_density = self.solid.solid_phase.density
+        solid_cp = self.solid.solid_phase.cp
+
+        # If this reactor has no left neighbor, it is the first reactor in the cascade.
+        # In this case, set initial mass fractions and enthalpies as inlet conditions
+        if self.neighbor_left is None:
+            Y_in = Y_in_inlet
+            h_in = h_in_inlet
+        else:  # otherwise, take the mass fraction and enthalpy from the left neighbor
+            Y_in = self.neighbor_left.thermo.Y
+            h_in = self.neighbor_left.thermo.enthalpy_mass
+
+        # ============================================================================#
+        #                          Temperature of the solid                           #
+        # ============================================================================#
+
+        LHS[self.index_Ts] = (1.0 - porosity) * solid_cp * solid_density * self.V
+
+        # heat transfer between gas and solid
+        RHS[self.index_Ts] += self.htc() * (self.thermo.T - self.Ts) * self.V
+
+        # compute the axial heat loss due to radiation to the ambience
+        # if this is the first or last reactor in the cascade
+        if self.neighbor_left is None or self.neighbor_right is None:
+            RHS[self.index_Ts] -= ct.stefan_boltzmann * self.solid.emissivity \
+                * (self.Ts**4 - Tamb**4) * self.A
+
+        # compute heat flux in the solid between the reactors in the cascade
+        if self.neighbor_left is not None:
+            RHS[self.index_Ts] -= heat_flux(self.neighbor_left, self) * self.A
+        if self.neighbor_right is not None:
+            RHS[self.index_Ts] += heat_flux(self, self.neighbor_right) * self.A
+
+        # ============================================================================#
+        #                             species mass fractions                          #
+        # ============================================================================#
+
+        wdot = self.kinetics.net_production_rates # chemical source terms
+        if not self.chemistry:
+            wdot *= 0.0
+
+        # right hand side of the mass fraction equations
+        for k in range(self.thermo.n_species):
+            index = self.component_index(self.thermo.species_name(k))
+            # chemical source term
+            RHS[index] += wdot[k] * self.thermo.molecular_weights[k] / density
+            # convective contribution
+            RHS[index] += self.A * mdot / \
+                (self.V * porosity * density) * (Y_in[k] - self.thermo.Y[k])
```

Comment:
Same thing applies here for `self.thermo.Y`, though that of course needs to be evaluated within `replace_eval`.